### ISSUE TITLE
_parse_url not producing the correct report detail URLs

### ISSUE THREAD

**Issue by jonrobinson2** [openelections/clarify#14]
Here is my simple script to try and pull unofficial precinct results from the 2015 governors race from the KY Clarity site.

``` python
import clarify
import wget
import os
import zipfile

import zipfile
import clarify
import wget
import os

j = clarify.Jurisdiction(url='http://results.enr.clarityelections.com/KY/15261/30235/en/summary.html', level='state')
subs=j.get_subjurisdictions()

name='detail'
end='.zip'

def getkyresults():
for x in range(0,len(subs)):
    print subs[x].report_url('txt')
    wget.download(subs[x].report_url('txt'))
    hm=zipfile.ZipFile('detailtxt.zip')
    zipfile.ZipFile.extractall(hm)
    os.rename('detailtxt.zip', name+str(x)+end)
    os.rename('detail.txt', name+str(x)+'.txt')

```

You can see that file 4 is a BadZipfile. This is because its a link to a 404 page.

**Comment by dwillis** [comment]
I'm getting a 404 for Bath County, which is the second file. So it looks like clarify needs to handle 404s better, but also that not all KY counties have produced detail files. Here's [Bath's detail page](http://results.enr.clarityelections.com/KY/Bath/15268/27411/en/reports.html) compared to [Adair](http://results.enr.clarityelections.com/KY/Adair/15263/27401/en/reports.html), which has the detail reports.

**Comment by jonrobinson2** [comment]
Hm, it seems like there are results though:

**Comment by jonrobinson2** [comment]
Gr, realizing I didn't put in the 2015 general URL. Anyways, when I do

```python
j = clarify.Jurisdiction(url='http://results.enr.clarityelections.com/KY/57510/156897/en/summary.html', level='state')
subs=j.get_subjurisdictions()
print subs
```

That call to print subs produces nothing. Something is funky w/ the subjurisdictions call for this formatting it looks like.

**Comment by dwillis** [comment]
There are summary results for Bath County, but the subjurisdictions are for precincts, and Bath hasn't reported any precinct-level results.

**Comment by jonrobinson2** [comment]
Hm, am I misreading the link? I click on that home icon and see precinct results.

**Comment by dwillis** [comment]
No, you're right. Hmm. I'll dig into why the URLs aren't right.

**Comment by jonrobinson2** [comment]
Thanks Derek.

**Comment by jonrobinson2** [comment]
Also, not sure the title of this issue is correct given your edit. You want to change it?

**Comment by dwillis** [comment]
yep

**Comment by dwillis** [comment]
Following up on this: I've added some code so that clarify does return the proper urls, but there are some instances (in KY for 2015, for example) where a county does not have detailed files. We have two options as I see it: 

1. Have clarify check the URL to see if it's good and, if not, return None as URL.
2. Leave URL checking to the user, like so:

```python
import zipfile
import clarify
import requests
import os

j = clarify.Jurisdiction(url='http://results.enr.clarityelections.com/KY/57510/156897/Web01/en/summary.html', level='state')
subs=j.get_subjurisdictions()

name='detail'
end='.zip'

for x in range(0,len(subs)):
    r = requests.get(subs[x].report_url('txt'))
    if r.status_code == 200:
        with open('detailtxt.zip', 'wb') as f:
            f.write(r.content)
        hm=zipfile.ZipFile('detailtxt.zip')
        zipfile.ZipFile.extractall(hm)
        os.rename('detailtxt.zip', name+str(x)+end)
        os.rename('detail.txt', name+str(x)+'.txt')
    else:
        print subs[x].name
```

I'm slightly leaning towards having clarify check the urls, but we don't do that in other cases. Thoughts?

**Comment by jonrobinson2** [comment]
Selfishly I'd love for you guys to check the urls!

**Comment by dwillis** [comment]
We'll do that, then. Thanks!

**Comment by jonrobinson2** [comment]
You rock

**Comment by GPHemsley** [comment]
Fixed in 69fffc593be477bc44cbaa8ee908cc097508ae63 and cfe3711359a43bc12bdc677a9d82c52fe15f326c.
